### PR TITLE
Add TS defs for the window management  API

### PR DESCRIPTION
**Pull request by AndreyBelym**
Add definitions for the multiwindow API. Descriptions are taken from the pending PR to the docs: #5363.

### CODE REVIEW

File: ts-defs-src/test-api/test-controller.d.ts [hunk 1]
```diff
@@ -82,6 +82,20 @@ interface Browser {
     prettyUserAgent: string;
 }
 
+type WindowSelector = unknown;
+
+interface SwitchToWindowPredicateData {
```
Comment (miherlosev) [inline ts-defs-src/test-api/test-controller.d.ts:87]: I propose to rename this interface as `SwithToWindowPredicate`.

File: ts-defs-src/test-api/test-controller.d.ts [hunk 2]
```diff
@@ -82,6 +82,20 @@ interface Browser {
     prettyUserAgent: string;
 }
 
+type WindowSelector = unknown;
```
Comment (miherlosev) [inline ts-defs-src/test-api/test-controller.d.ts:85]: I think the `WindowDescriptor` and `WindowDescriptorPromise` are more suitable names.

File: ts-defs-src/test-api/test-controller.d.ts (hunk 1 above)
Comment (AndreyBelym) [reply ts-defs-src/test-api/test-controller.d.ts:87]: `SwitchToWindowPredicate` is a function, this type represents its argument.

Comment (AndreyBelym) [reply ts-defs-src/test-api/test-controller.d.ts:87]: We can use `SwitchToWindowPredicateArgument` instead.

File: ts-defs-src/test-api/test-controller.d.ts (hunk 2 above)
Comment (AndreyBelym) [reply ts-defs-src/test-api/test-controller.d.ts:85]: We use it in the code: https://github.com/DevExpress/testcafe/blob/87094b31c74d1ea20b3286f2e07dcaef3d87a5df/src/api/test-controller/index.js#L320

I just wanted to be consistent with the code. I agree that `WindowDescriptor` can be better, but we then should rename corresponding names in the code too. @AlexKamaev, do you agree?

Comment (AlexKamaev) [reply ts-defs-src/test-api/test-controller.d.ts:85]: Yes, I agree that WindowDescriptor is more suitable and not so confusing as Selector

File: ts-defs-src/test-api/test-controller.d.ts (hunk 1 above)
Comment (AlexKamaev) [reply ts-defs-src/test-api/test-controller.d.ts:87]: SwitchToWindowPredicateArgs?

Comment (miherlosev) [reply ts-defs-src/test-api/test-controller.d.ts:87]: I tried to search the `PredicateData`, `PredicateArgument` and `PredicateArgs` names in https://github.com/DefinitelyTyped/DefinitelyTyped repository and I did not find such names.
The predicate data argument has a name that describes the entity passed there.

So, I propose the `WindowData` or `WindowPartialData`.

File: ts-defs-src/test-api/test-controller.d.ts (hunk 2 above)
Comment (miherlosev) [reply ts-defs-src/test-api/test-controller.d.ts:85]: Ok

File: ts-defs-src/test-api/test-controller.d.ts (hunk 1 above)
Comment (AndreyBelym) [reply ts-defs-src/test-api/test-controller.d.ts:87]: `Args` in different forms used in DefinitelyTyped:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5344bfc80508c53a23dae37b860fb0c905ff7b24/types/wampy/index.d.ts#L15

Comment (AndreyBelym) [reply ts-defs-src/test-api/test-controller.d.ts]: `WindowData` is a bit obscure. What do you think about `WindowFilterData`?

Comment (miherlosev) [reply ts-defs-src/test-api/test-controller.d.ts]: Ok, let it be.

**Pull request merged**